src/storage/src/hummock/event_handler/uploader.rs

```diff
@@ -297,13 +381,56 @@ impl SealedData {
         self.spilled_data.uploaded_data = unseal_epoch_data.spilled_data.uploaded_data;
     }
 
+    fn add_merged_imm(&mut self, merged_imm: &ImmutableMemtable) {
+        debug_assert!(merged_imm.is_merged_imm());
+        // add merged_imm to merged_imms
+        self.merged_imms.push_front(merged_imm.clone());
+    }
+
+    fn drop_merging_tasks(&mut self) {
+        // scan from oldest merging task to restore
+        // merge candidate imms back
+        for task in self.merging_tasks.iter().rev() {
```

Comment:
Shall we introduce an assumption that a merging task is only for a specific table id and instance id? If so we can avoid repeatedly calling `task.input_imms.entry(...)`